community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/csv/EagerlyReadInputChunk.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.unsafe.impl.batchimport.input.csv;
+
+import java.io.IOException;
+
+import org.neo4j.csv.reader.Source;
+import org.neo4j.unsafe.impl.batchimport.input.InputChunk;
+import org.neo4j.unsafe.impl.batchimport.input.InputEntity;
+import org.neo4j.unsafe.impl.batchimport.input.InputEntityVisitor;
+
+class EagerlyReadInputChunk implements InputChunk, Source.Chunk
+{
+    private InputEntity[] entities;
+    private int cursor;
+
+    void initialize( InputEntity[] entities )
+    {
+        this.entities = entities;
+        this.cursor = 0;
+    }
+
+    @Override
+    public boolean next( InputEntityVisitor visitor ) throws IOException
+    {
+        if ( cursor < entities.length )
+        {
+            entities[cursor++].replayOnto( visitor );
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+    }
+
+    @Override
+    public char[] data()
+    {
+        throw new UnsupportedOperationException();
```

Comment:
It's this weird thing where there's the normal lazy chink/parser and then we have to, against our will, support the multi-line fields feature and it's conflation of newline with record delimiters... this changes how parsing can be done significantly.

I'd guess this is a turd from that behind. I'm sure it can be done better, I just don't see it clearly a.t.m.